src/mpp/Dec.hpp

```diff
@@ -936,6 +993,32 @@ bool jump_read(BUF& buf, T... t)
 
 decode_next_label:
 	return decode_next<PATH>(buf, t...);
+
+}
+
+template <compact::Family FAMILY, size_t SUBRULE,
+	  class PATH, class BUF, class... T>
+bool jump_read(BUF& buf, T... t)
+{
+	auto&& dst = unwrap(path_resolve_transparent(PATH{}, t...));
+	return jump_read_impl<FAMILY, SUBRULE, PATH>(buf, dst, t...);
+}
+
+template <compact::Family FAMILY, size_t SUBRULE,
+	  class PATH, class BUF, class... T>
+bool jump_read_optional(BUF& buf, T... t)
+{
+	auto&& optional_dst = unwrap(path_resolve(PATH{}, t...));
+	if constexpr (FAMILY == compact::MP_NIL) {
+		[[maybe_unused]] auto val = read_value<FAMILY, SUBRULE>(buf);
+		optional_dst.reset();
+		return decode_next<PATH>(buf, t...);
+	} else {
+		if (!optional_dst.has_value())
+			optional_dst.emplace();
```

Comment:
Could you elaborate the purpose of this step?

Reply:
Later, we want to use underlying value directly. For this purpose, we should emplace `optional` is it's empty.